test/PizzeriaTest.cpp

```diff
@@ -28,40 +28,78 @@ TEST_F(PizzeriaTest, priceForMargherita25AndFunghi30ShouldBe55)
     ASSERT_EQ(55, price);
 }
 
-TEST_F(PizzeriaTest, bakeDummyPizza)
-{
+TEST_F(PizzeriaTest, bakeDummyPizza) {
     // Given
     Pizzas pizzas = {new PizzaDummy{}};
+    EXPECT_CALL(mt, sleep_for(minutes(0))).Times(1);
 
     // When
     auto orderId = pizzeria.makeOrder(pizzas);
     pizzeria.bakePizzas(orderId);
 }
 
-TEST_F(PizzeriaTest, completeOrderWithStubPizza)
-{
+TEST_F(PizzeriaTest, completeOrderWithStubPizza) {
     // Given
     Pizzas pizzas = {new PizzaStub{"STUB"}};
+    EXPECT_CALL(mt, sleep_for(minutes(1))).Times(1);
 
     // When
     auto orderId = pizzeria.makeOrder(pizzas);
     pizzeria.bakePizzas(orderId);
     pizzeria.completeOrder(orderId);
 }
 
-TEST_F(PizzeriaTest, calculatePriceForPizzaMock)
-{   
+TEST_F(PizzeriaTest, calculatePriceForPizzaMock) {
     // Given
     PizzaMock* mock = new PizzaMock{};
     Pizzas pizzas = {mock};
     EXPECT_CALL(*mock, getPrice()).WillOnce(Return(40.0));
-    
+
     // When
     auto orderId = pizzeria.makeOrder(pizzas);
+
     auto price = pizzeria.calculatePrice(orderId);
 
     // Then
     ASSERT_EQ(40, price);
 
     delete mock;
 }
+
+TEST_F(PizzeriaTest, calculateOrderLikeInMainForStubPizzaAndTwoMockPizza) {
+    // Given
+    PizzaStub stubPizza{"stubPizza"};
+    StrictMock<PizzaMock> strictMockPizza{};
+    NiceMock<PizzaMock> niceMockPizza{};
+
+    Pizzas pizzas = {&stubPizza, &strictMockPizza, &niceMockPizza};
+
+    constexpr double nicePizzaPrice = 18.0;
+    constexpr double strictPizzaPrice = 12.5;
+    constexpr char nicePizzaName[] = "nicePizza";
+    constexpr char strictPizzaName[] = "StrictPizza";
+    constexpr minutes niceBakingTime{2};
+    constexpr minutes strictBakingTime{3};
+
+    EXPECT_CALL(mt, sleep_for(minutes(1))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(2))).Times(1);
+    EXPECT_CALL(mt, sleep_for(minutes(3))).Times(1);
```

Comment:
Use predefined constexpr values:
```suggestion
    EXPECT_CALL(mt, sleep_for(stubPizza->getBakingTime())).Times(1);
    EXPECT_CALL(mt, sleep_for(niceBakingTime)).Times(1);
    EXPECT_CALL(mt, sleep_for(strictBakingTime)).Times(1);
```